feat(editor): :read reads the output of an Ex command into the buffer

benarcher2691 opened the pull request:
## Summary

- `:read :cmd` now reads the output of an Ex command into the buffer
- Uses `put=execute(cmd)` internally to capture and insert command output

## Original Work

This continues the work from #30628 by @bwalshe, preserving their authorship as requested.

## Test Plan

- [x] All existing tests pass
- [x] New tests in `test/functional/ex_cmds/read_spec.lua`:
  - Inserts text from Ex command
  - Inserts at specific position with line number
  - Handles command chaining with `|`
  - Undo works correctly
  - Error handling for invalid commands

bwalshe committed:
feat(editor): :read reads the output of an Ex command into the buffer

Problem: :read can read from files or stdin, but not from other Ex commands

Solution: use `put=execute(cmd)` to execute cmd and send its output to the
buffer

glepnir commented on src/nvim/ex_docmd.c:
```diff
@@ -6010,6 +6012,32 @@ static void ex_syncbind(exarg_T *eap)
+  do_cmdline(put_cmd.items, eap->ea_getline, eap->cookie, eap->flags);
+  kv_destroy(put_cmd);
+  nvim_del_var(cmd_var_name, &error);
+  if (error.type != kErrorTypeNone) {
```
```suggestion
  if (!ERROR_SET(error)) {
```

benarcher2691 commented:
Thanks for the review @glepnir! I've addressed the feedback:

- Using `ERROR_SET(&error)` for error checking
- Calling `api_clear_error(&error)` after `emsg()`

Ready for another look when you have a chance.
